### DISCUSSION TITLE
Invalid file type (xml) during upload

### DISCUSSION THREAD

**Discussion by ghubz** [WeblateOrg/weblate#6893] [Q&A] [answered]
Hi,

Setup:
``` 
* Weblate: 4.9.1
 * Django: 3.2.9
 * siphashc: 2.1
 * translate-toolkit: 3.5.1
 * lxml: 4.6.4
 * Pillow: 8.4.0
 * bleach: 4.1.0
 * python-dateutil: 2.8.2
 * social-auth-core: 4.1.0
 * social-auth-app-django: 5.0.0
 * django-crispy-forms: 1.13.0
 * oauthlib: 3.1.0
 * django-compressor: 2.4.1
 * djangorestframework: 3.12.4
 * django-filter: 21.1
 * django-appconf: 1.0.5
 * user-agents: 2.2.0
 * filelock: 3.4.0
 * setuptools: 59.4.0
 * jellyfish: 0.8.9
 * openpyxl: 3.0.9
 * celery: 5.2.1
 * kombu: 5.2.2
 * translation-finder: 2.10
 * weblate-language-data: 2021.6
 * html2text: 2020.1.16
 * pycairo: 1.20.1
 * pygobject: 3.42.0
 * diff-match-patch: 20200713
 * requests: 2.26.0
 * django-redis: 5.1.0
 * hiredis: 2.0.0
 * sentry_sdk: 1.5.0
 * Cython: 0.29.24
 * misaka: 2.1.1
 * GitPython: 3.1.24
 * borgbackup: 1.1.17
 * pyparsing: 3.0.6
 * pyahocorasick: 1.4.2
 * python-redis-lock: 3.7.0
 * Python: 3.7.3
 * Git: 2.20.1
 * psycopg2-binary: 2.8.6
 * phply: 1.2.5
 * chardet: 4.0.0
 * ruamel.yaml: 0.17.4
 * tesserocr: 2.5.1
 * akismet: 1.1
 * boto3: 1.17.53
 * zeep: 4.0.0
 * aeidon: 1.9
 * iniparse: 0.5
 * Mercurial: 5.7.1
 * Redis server: 5.0.3
 * PostgreSQL server: 11.14
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: localhost
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://localhost:6379, redis://localhost:6379, regular
 ```

I'm trying to upload a translation file named en.xml (I did this multiple times in the past and it was fine) but now I get that the file type is invalid with **Unknown filetype (en.xml)** . In sentry I see the stack trace and it shows the following, which apparently somehow does not include xml:

`classes_str	
    {
    csv: [
    'csvl10n', 
    'csvfile'
    ], 
    gmo: [
    'mo', 
    'mofile'
    ], 
    lang: [
    'mozilla_lang', 
    'LangStore'
    ], 
    mo: [
    'mo', 
    'mofile'
    ], 
    po: [
    'po', 
    'pofile'
    ], 
    pot: [
    'po', 
    'pofile'
    ], 
    qm: [
    'qm', 
    'qmfile'
    ], 
    tab: [
    'omegat', 
    'OmegaTFileTab'
    ], 
    utf8: [
    'omegat', 
    'OmegaTFile'
    ], 
    utx: [
    'utx', 
    'UtxFile'
    ]
}`

I don't know if an update broke this and I also cannot find anywhere that this is configurable. Flat XML should be enabled by default. 

Any ideas?

**Comment by nijel** [comment]
In case some file format is not supported, you should get an alert in the management interface (see https://docs.weblate.org/en/latest/admin/install.html#production-setup) or in deploy checks.

**Comment by ghubz** [comment]
The issue is, the file format IS supported. It is flat XML. It is the same format/file I have been uploading for quite some time now. I even download the existing en.xml file from weblate and reupload it and get the same error.. (**File upload has failed: Unknown filetype (en.xml)**)

Any ideas why would `classes_str { csv: [ 'csvl10n', 'csvfile' ], gmo: [ 'mo', 'mofile' ], lang: [ 'mozilla_lang', 'LangStore' ], mo: [ 'mo', 'mofile' ], po: [ 'po', 'pofile' ], pot: [ 'po', 'pofile' ], qm: [ 'qm', 'qmfile' ], tab: [ 'omegat', 'OmegaTFileTab' ], utf8: [ 'omegat', 'OmegaTFile' ], utx: [ 'utx', 'UtxFile' ] }` not include XML?

Deploy checks don't yield anything. 

It only has the following which is expected: 

```WARNINGS:
?: (weblate.W025.fluent) Failure in loading handler for fluent file format: No module named 'fluent'
        HINT: https://docs.weblate.org/en/latest/admin/install.html#optional-deps
?: (weblate.W033.Gerrit) Failure in loading VCS module for Gerrit: git: 'review' is not a git command. See 'git --help'.
 (1)
        HINT: https://docs.weblate.org/en/latest/vcs.html
?: (weblate.W033.Subversion) Failure in loading VCS module for Subversion: git: 'svn' is not a git command. See 'git --help'.

The most similar command is
        serve
 (1)
        HINT: https://docs.weblate.org/en/latest/vcs.html

System check identified 3 issues (1 silenced).```

**Reply by ghubz** [reply]
P.S: Uploading the same file as part of a zip in a new component works fine.

**Reply by nijel** [reply]
`classes_str` is expected to not contain it, it is used only for some formats. The root cause is elsewhere - this is last resort autodetection and the file format configured on the component should be tried first. For some reason the logic did not work. The related code is here:

https://github.com/WeblateOrg/weblate/blob/2ab842939b4cdb1ec024820c8ed4c7b2a858f3d4/weblate/formats/auto.py#L41-L84

**Comment by nijel** [comment, marked as answer]
In case you happen to upload to add new strings, this should be addressed by 759edc89e7.

**Reply by ghubz** [reply]
That's exactly what I was trying to do! Thank you!

**Reply by ghubz** [reply]
Btw can I make those changes to the 2 files manually until the next release is out? Just so that I can continue with my work.

**Reply by nijel** [reply]
Yes, but the release should be out today or tomorrow :-).

**Reply by ghubz** [reply]
Super, thanks

**Reply by ghubz** [reply]
I confirm that the issue has been fixed after upgrading to 4.10.
